src/anemoi/inference/inputs/ekd.py

```diff
@@ -128,298 +106,61 @@ def __init__(
         self._namer = namer if namer is not None else self.checkpoint.default_namer()
         assert callable(self._namer), type(self._namer)
 
-    def _filter_and_sort(self, data: Any, *, variables: List[str], dates: List[Any], title: str) -> Any:
-        """Filter and sort the data.
+    def create_state(self, *, date=None, variables=None, initial: bool = True) -> State:
+        # NOTE: all logic involing the checkpoint should take place here
 
-        Parameters
-        ----------
-        data : Any
-            The data to filter and sort.
-        variables : List[str]
-            The list of variables to select.
-        dates : List[Any]
-            The list of dates to select.
-        title : str
-            The title for logging.
+        # define request (using checkpoint)
+        variables = self.checkpoint_variables if variables is None else variables
```

Comment:
Using `self.checkpoint_variables` here is wrong. That one contains all input variables including the computed forcings. The state here should be created only from non-computed variables, since they are added to it afterwards.

You need to use:
`self.checkpoint.variables_from_input(include_forcings=True)`
This one skips computed forcings.